transformations/correct_common_misspellings/transformation.py

```diff
@@ -0,0 +1,4112 @@
+import spacy
+
+from initialize import spacy_nlp
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class CorrectCommonMisspellings(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+    keywords = [
+        "lexical",
+        "external-knowledge-based",
+        "tokenizer-required",
+        "highly-meaning-preserving",
+        "high-coverage",
+        "low-generations",
+    ]
+
+    def __init__(self):
+        super().__init__()
+        self.COMMON_MISSPELLINGS_DICT = get_common_misspellings_dict()
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+
+    def generate(self, sentence: str):
+        doc = self.nlp(sentence)
+        perturbed_text = [
+            self.COMMON_MISSPELLINGS_DICT.get(token.text, token.text) + " "
+            if token.whitespace_
+            else self.COMMON_MISSPELLINGS_DICT.get(token.text, token.text)
+            for token in doc
+        ]
+        return ["".join(perturbed_text)]
+
+
+def get_common_misspellings_dict():
```

Comment:
Can this be a separate json file or so that gets loaded here?

Reply:
Thanks good idea -- I just moved the dictionary into a separate json file which gets loaded!